src/graph.jl

```diff
@@ -44,6 +44,60 @@ function Graph{T<:Number}(adjmx::Array{T, 2})
     return g
 end
 
+#function convert(::Type{Vector{Vector{T}}}, a::SparseMatrixCSC{T})
+function raggedize(a::SparseMatrixCSC)
+    numcols = size(a,2)
+    cols = Array(Vector{Int}, numcols)
+    for i=1:numcols
+        rangidx = a.colptr[i]:a.colptr[i+1]-1
+        cols[i] = a.rowval[rangidx]
+    end
+    return cols
+end
+
+function getfinclist(neighborhood, i)
+    degi = length(neighborhood)
+    edges = Array(Edge, degi)
+    for j = 1:degi
+        edges[j] = Edge(i, neighborhood[j])
+    end
+    return edges
+end
+
+function getbinclist(neighborhood, i)
+    degi = length(neighborhood)
+    edges = Array(Edge, degi)
+    for j = 1:degi
+        edges[j] = Edge(neighborhood[j], i)
+    end
+    return edges
+end
+
+function Graph{T<:Number}(adjmx::SparseMatrixCSC{T})
+    numrows = size(adjmx,1)
+    numcols = size(adjmx,2)
+    numrows == numcols || error("Adjacency matrices must be square: $numrows == $numcols")
+    finclist = Array(Vector{Edge}, numrows)
+    binclist = Array(Vector{Edge}, numrows)
+    g = Graph(1:numrows, Set{Edge}(), binclist, finclist)
+    cols = raggedize(adjmx)
+    println("pushing edges")
```

Comment:
That print and the following `@time` is there to show that the time taken by the set operations accounts for the majority of the total time for this function. You can see it in the `test/perf/graphfromCSC.jl` file